### DISCUSSION TITLE
"always name threads" guideline?

### DISCUSSION THREAD

**Discussion by Ben-PH** [massalabs/massa#3604] [Code Style]
instead of

```rust
std::thread::spawn(...);
```
use

```rust
std::thread::Builder::new().name("useful-name-here").spawn(...).unwrap();
```

Not so sure why a builder-based spawn gives a Result. Not so sure what a good pattern for handling a builder-spawn error might be.

What are your thoughts?

**Comment by AurelienFT** [comment]
Yes. We should be more attentive at review but naming thread must be a convention in our code.
